Extensions/Lighting/lightobstacleruntimebehavior.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @memberof gdjs
+ * @class LightObstaclesManager
+ * @param {gdjs.RuntimeScene} runtimeScene
+ */
+gdjs.LightObstaclesManager = function (runtimeScene) {
+  this._obstacleRBush = new rbush(9, [
+    '.owner.getAABB().min[0]',
+    '.owner.getAABB().min[1]',
+    '.owner.getAABB().max[0]',
+    '.owner.getAABB().max[1]',
+  ]);
+};
+
+gdjs.LightObstaclesManager.getManager = function (runtimeScene) {
+  if (!runtimeScene.lightObstaclesManager) {
+    // Create the shared manager if necessary.
+    runtimeScene.lightObstaclesManager = new gdjs.LightObstaclesManager(
+      runtimeScene
+    );
+  }
+
+  return runtimeScene.lightObstaclesManager;
+};
+
+gdjs.LightObstaclesManager.prototype.addObstacle = function (obstacle) {
+  this._obstacleRBush.insert(obstacle);
+};
+
+gdjs.LightObstaclesManager.prototype.removeObstacle = function (obstacle) {
+  this._obstacleRBush.remove(obstacle);
+};
+
+gdjs.LightObstaclesManager.prototype.getAllObstaclesAround = function (
+  object,
+  radius,
+  result
+) {
+  // TODO: This would better be done using the object AABB (getAABB), as (`getCenterX`;`getCenterY`) point
+  // is not necessarily in the middle of the object (for sprites for example).
+  var x = object.getX();
+  var y = object.getY();
+
+  var searchArea = gdjs.staticObject(
+    gdjs.LightObstaclesManager.prototype.getAllObstaclesAround
+  );
+  searchArea.minX = x - radius;
+  searchArea.minY = y - radius;
+  searchArea.maxX = x + radius;
+  searchArea.maxY = y + radius;
+  var nearbyPlatforms = this._obstacleRBush.search(searchArea);
+  result.length = 0;
+  result.push.apply(result, nearbyPlatforms);
+};
+
+/**
+ * @memberof gdjs
+ * @class LightObstacleRuntimeBehavior
+ * @param {gdjs.RuntimeScene} runtimeScene
+ * @param {*} behaviorData
```

Comment:
Prefer to use `any` than `*`, it's more explicit and TypeScript compatible.
In this case, you can use the type `BehaviorData`, that is a global type declared in `runtimebehavior.js`